ir/instr.h

```diff
@@ -399,13 +399,17 @@ class Alloc final : public Instr {
 
 
 class Malloc final : public Instr {
-  Value *size;
+  Value *ptr, *size;
   // Is this malloc (or equivalent operation, like new()) never returning
   // null?
   bool isNonNull;
+  bool isRealloc;
+
 public:
-  Malloc(Type &type, std::string &&name, Value &size, bool isNonNull)
-    : Instr(type, std::move(name)), size(&size), isNonNull(isNonNull) {}
+  Malloc(Type &type, std::string &&name, Value &ptr, Value &size,
```

Comment:
what about adding two constructors, one for malloc (single value), and another for realloc (2 values)?